### PR TITLE
Feature/ddbteam 1040 openlibrary vendor

### PR DESCRIPTION
**Pull request by turegjorup**
#### Link to ticket

https://jira.itkdev.dk/browse/DDBTEAM-1040

#### Description

Depends on: https://github.com/danskernesdigitalebibliotek/ddb-cover-service-importers/pull/125

* Refactor vendor interfaces to allow vendor with no import support
* Renamed `SupportsSingleIdentifierInterface`to `VendorServiceSingleIdentifierInterface` for better naming consistency
* Added OpenLibrary vendor

#### Screenshot of the result

If your change affects the user interface you should include a screenshot of the result with the pull request.

#### Checklist

- [x] My code is covered by test cases.
- [x] My code passes our test (all our tests).
- [x] My code passes our static analysis suite.
- [x] My code passes our continuous integration process.

If your code does not pass all the requirements on the checklist you have to add a comment explaining why this change 
should be exempt from the list.

#### Additional comments or questions

If you have any further comments or questions for the reviewer please add them here.

### CODE REVIEW

File: src/Service/VendorService/OpenLibrary/OpenLibraryVendor.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    public function getUnverifiedVendorImageItem(string $identifier, string $type): UnverifiedVendorImageItem
+    {
+        if (!$this->supportsIdentifierType($type)) {
+            throw new UnsupportedIdentifierTypeException('Unsupported single identifier type: '.$type);
+        }
+
+        $item = new UnverifiedVendorImageItem();
+        $item->setIdentifier($identifier);
+        $item->setIdentifierType($type);
+        $item->setVendor($this->vendorCoreService->getVendor(self::VENDOR_ID));
```
Comment (cableman) [inline src/Service/VendorService/OpenLibrary/OpenLibraryVendor.php:36]: Here I would link to have the vendor into an variable, to make the code faster to read.

File: src/Service/VendorService/OpenLibrary/OpenLibraryVendor.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    {
+        return IdentifierType::ISBN === $type;
+    }
+
+    /**
+     * Get Vendors image URL from ISBN.
+     */
+    private function getVendorsImageUrl(string $isbn): string
+    {
+        return sprintf(self::COVER_URL_FORMAT, $isbn);
```
Comment (cableman) [inline src/Service/VendorService/OpenLibrary/OpenLibraryVendor.php:55]: Should we look into using https://github.com/thecodingmachine/safe ?

`use function safe/sprintf as sprintf`

File: src/Service/VendorService/UserUpload/UserUploadVendorService.php [hunk 3]
```diff
@@ -6,40 +6,15 @@
...
 /**
  * Class UserUploadVendorService.
  */
 class UserUploadVendorService implements VendorServiceInterface
 {
-    use ProgressBarTrait;
     use VendorServiceTrait;
 
     protected const VENDOR_ID = 15;
-
```
Comment (cableman) [inline src/Service/VendorService/UserUpload/UserUploadVendorService.php:25]: Why this change? and how do it relate to an new importer?

Comment (cableman) [reply src/Service/VendorService/UserUpload/UserUploadVendorService.php:25]: I see....

**Review by cableman — CHANGES_REQUESTED** [review]

File: src/Service/VendorService/OpenLibrary/OpenLibraryVendor.php (hunk 1 above)
Comment (turegjorup) [reply src/Service/VendorService/OpenLibrary/OpenLibraryVendor.php:36]: Fixed

File: src/Service/VendorService/OpenLibrary/OpenLibraryVendor.php (hunk 2 above)
Comment (turegjorup) [reply src/Service/VendorService/OpenLibrary/OpenLibraryVendor.php]: Changelog:
PHP 8: This function no longer returns false on failure. 
https://www.php.net/manual/en/function.sprintf.php and https://github.com/thecodingmachine/safe/blob/d48e85e101330b30b0c2555df063f0c719b98d7f/deprecated/strings.php#L296

So no reason to introduce `thecodingmachine/safe` here. But we can consider it for a later time.

File: src/Service/VendorService/UserUpload/UserUploadVendorService.php (hunk 3 above)
Comment (turegjorup) [reply src/Service/VendorService/UserUpload/UserUploadVendorService.php:25]: The refactoring of VendorServiceInterfaces means the `load()` function can be removed. Moving `getVendor()` to `VendorServiceTrait`made sense to me bacause it contains all other `getVendor*` functions.

**Review by cableman — APPROVED** [review]
